repotracker/github_poller.go

```diff
@@ -188,11 +188,22 @@ func (gRepoPoller *GithubRepositoryPoller) GetRevisionsSince(revision string, ma
 				"unable to find a suggested merge base commit for revision %v, must fix on projects settings page",
 				revision)
 			gRepoPoller.ProjectRef.RepotrackerError = revisionDetails
+			return []model.Revision{}, revisionError
 		} else {
-			// automatically set the newly found base revision as base revision
-			revisionError = errors.Errorf("base revision, %v not found, suggested base revision, %v found",
-				revision, baseRevision)
+			// automatically set the newly found base revision as base revision and append revisions
+			commit, err := thirdparty.GetCommitEvent(ctx,
+				gRepoPoller.OauthToken,
+				gRepoPoller.ProjectRef.Owner,
+				gRepoPoller.ProjectRef.Repo,
+				baseRevision,
+			)
+			if err != nil {
+				return nil, errors.Wrapf(err, "error loading commit '%v'", baseRevision)
+			}
+			revisions = append(revisions, githubCommitToRevision(commit))
 
+			grip.Infof("Base revision, %v not found, updating %v as new base revision",
```

Comment:
Logging directly like this isn't super useful, because we don't have a good way to search in Splunk. I suggest using message.Fields with the metadata split into different fields (the info log at the end of the function is an example of this).